frontend/markdown-pdf/build.js

```diff
@@ -113,77 +157,668 @@ Promise.all([
+        filePath.replace(/\.md$/, ''),
+        html,
+        css,
+        fp_css,
+        'A6',
+        {
+            top: '25mm',
+            right: '5mm',
+            left: '5mm',
+            bottom: '5mm',
+        }
+    );
+}
+
+/**
+ * This function is used to generate an empty PDF page, which can be used
+ * for the formatting of the booklet
+ * @param {*} browser Puppeteer browser instance
+ * @param {*} format Page size of the empty PDF page
+ */
+async function renderEmptyPage(browser, format = 'A4') {
+    // Create a page in the browser
+    const page = await browser
+        .newPage()
+        .then(page => page.emulateMedia('screen').then(() => page));
+
+    // Return a Buffer containing the empty page
+    return page.pdf({
+        format: format,
+    });
+}
+
+/**
+ * This function is used to generate a booklet version of the PDF of a manual,
+ * staring at page number `pageNumber`, given the HTML, CSS, (extra) booklet CSS,
+ * logo of Victron Energy, the current language used in the HTML, a list with all
+ * languages used in the booklet, and whether to display the sidebar
+ * @param {*} browser Puppeteer browser instance
+ * @param {*} title Title of the booklet that will be displayed at the bottom of the booklet
+ * @param {*} html HTML content of the booklet
+ * @param {*} css General CSS for the booklet
+ * @param {*} bookletCSS Extra CSS for the booklet
+ * @param {*} logoSVG SVG file containing the logo of Victron Energy
+ * @param {*} lang Language used in the HTML
+ * @param {*} languages All languages used in the booklet (necessary when display_sidebar is true)
+ * @param {*} display_sidebar Boolean indicating whether we should display the sidebar
+ * @param {*} pageNumber Pagenumber to start numbering from
+ */
+async function generateBooklet(
+    browser,
+    title,
+    html,
+    css,
+    bookletCSS,
+    logoSVG,
+    lang,
+    languages,
+    display_sidebar = true,
+    pageNumber = 1
+) {
+    // Generate the HTML for the content of the booklet
+    const content = `
+    <!DOCTYPE html>
+    <html>
+    <head>
+        <style>${css}</style>
+        <style>${bookletCSS}</style>
+    </head>
+    <body>
+        ${
+            /* Add `pageNumber` empty pages to the HTML in order to increase the page number */
+            [...Array(pageNumber).keys()].map(
+                () => `<div class="empty-page">empty</div>`
+            )
+        }
+        <div class="">
+            <div class="sidebar" ${
+                /* Set display to 'none', depending on `display_sidebar` */
+                display_sidebar
+                    ? 'style=""'
+                    : 'style="display: none !important"'
+            }>
+                <table class="sidebar">
+                    <tr>
+                        ${
+                            /* Create a side bar from `languages`, where `lang` is selected */
+                            languages
+                                .map(lang_ => {
```

Comment:
Kunnen we `lang` hernoemen naar `currentLang` oid om de leesbaarheid van dit stukje code te verbeteren?